lib/Interpreter/CXCppInterOp.cpp

```diff
@@ -0,0 +1,627 @@
+    return CXCursor_EnumConstantDecl;
+  case Decl::Field:
+    return CXCursor_FieldDecl;
+  case Decl::Function:
+    return CXCursor_FunctionDecl;
+  case Decl::CXXMethod:
+    return CXCursor_CXXMethod;
+  case Decl::CXXConstructor:
+    return CXCursor_Constructor;
+  case Decl::CXXDestructor:
+    return CXCursor_Destructor;
+  case Decl::CXXConversion:
+    return CXCursor_ConversionFunction;
+  case Decl::ParmVar:
+    return CXCursor_ParmDecl;
+  case Decl::Typedef:
+    return CXCursor_TypedefDecl;
+  case Decl::TypeAlias:
+    return CXCursor_TypeAliasDecl;
+  case Decl::TypeAliasTemplate:
+    return CXCursor_TypeAliasTemplateDecl;
+  case Decl::Var:
+    return CXCursor_VarDecl;
+  case Decl::Namespace:
+    return CXCursor_Namespace;
+  case Decl::NamespaceAlias:
+    return CXCursor_NamespaceAlias;
+  case Decl::TemplateTypeParm:
+    return CXCursor_TemplateTypeParameter;
+  case Decl::NonTypeTemplateParm:
+    return CXCursor_NonTypeTemplateParameter;
+  case Decl::TemplateTemplateParm:
+    return CXCursor_TemplateTemplateParameter;
+  case Decl::FunctionTemplate:
+    return CXCursor_FunctionTemplate;
+  case Decl::ClassTemplate:
+    return CXCursor_ClassTemplate;
+  case Decl::AccessSpec:
+    return CXCursor_CXXAccessSpecifier;
+  case Decl::ClassTemplatePartialSpecialization:
+    return CXCursor_ClassTemplatePartialSpecialization;
+  case Decl::UsingDirective:
+    return CXCursor_UsingDirective;
+  case Decl::StaticAssert:
+    return CXCursor_StaticAssert;
+  case Decl::Friend:
+    return CXCursor_FriendDecl;
+  case Decl::TranslationUnit:
+    return CXCursor_TranslationUnit;
+
+  case Decl::Using:
+  case Decl::UnresolvedUsingValue:
+  case Decl::UnresolvedUsingTypename:
+    return CXCursor_UsingDeclaration;
+
+  case Decl::UsingEnum:
+    return CXCursor_EnumDecl;
+
+  default:
+    if (const auto* TD = dyn_cast<TagDecl>(D)) {
+      switch (TD->getTagKind()) {
+#if CLANG_VERSION_MAJOR >= 18
+      case TagTypeKind::Interface: // fall through
+      case TagTypeKind::Struct:
+        return CXCursor_StructDecl;
+      case TagTypeKind::Class:
+        return CXCursor_ClassDecl;
+      case TagTypeKind::Union:
+        return CXCursor_UnionDecl;
+      case TagTypeKind::Enum:
+        return CXCursor_EnumDecl;
+#else
+      case TagTypeKind::TTK_Interface: // fall through
+      case TagTypeKind::TTK_Struct:
+        return CXCursor_StructDecl;
+      case TagTypeKind::TTK_Class:
+        return CXCursor_ClassDecl;
+      case TagTypeKind::TTK_Union:
+        return CXCursor_UnionDecl;
+      case TagTypeKind::TTK_Enum:
+        return CXCursor_EnumDecl;
+#endif
+      }
+    }
+  }
+
+  return CXCursor_UnexposedDecl;
+}
+
+CXTypeKind cxtype_GetBuiltinTypeKind(const BuiltinType* BT) {
+#define BTCASE(K)                                                              \
+  case BuiltinType::K:                                                         \
+    return CXType_##K
+  switch (BT->getKind()) {
```

Comment:
Can't we use something like:

```
define BUILTIN_TYPE(Id, SingletonId) \
  case BuiltinType::Id: return ...
#include "clang/AST/BuiltinTypes.def"
```

Reply:
These are copied from libclang's source code. Since we will make the merge in the upstream in the future, it would be better to submit PRs to the upstream directly.